packages/atomic/src/components/common/generated-answer/generated-content/markdownUtils.ts

```diff
@@ -0,0 +1,59 @@
+import {marked} from 'marked';
+
+const unclosedItalic = /(?:\*)(\w[\w\s]*)$/;
+const unclosedStrong = /(?:\*{2})(\w[\w\s]*)$/;
+const unclosedStrongItalic = /(?:\*{3})(\w[\w\s]*)$/;
+const unclosedCode = /(?:`)(\w[\w\s]*)$/;
+
+const completeUnclosedElement = (text: string) => {
+  if (unclosedStrongItalic.test(text)) {
+    return text.replace(unclosedStrongItalic, '<strong><em>$1</em></strong>');
+  }
+  if (unclosedStrong.test(text)) {
+    return text.replace(unclosedStrong, '<strong>$1</strong>');
+  }
+  if (unclosedItalic.test(text)) {
+    return text.replace(unclosedItalic, '<em>$1</em>');
+  }
+  if (unclosedCode.test(text)) {
+    return text.replace(unclosedCode, '<code>$1</code>');
+  }
+
+  return text;
+};
```

Comment:
I am curious about the usage of the many `return` s.

Looking at the function layout. I assume that the text will never have both a unclosed italic and a unclosed strong at the same time. Since the function will stop its execution at the italic condition.
So I guess my question is more "what does the _text_ parameter means ?", its a bit vague. is it a complete paragraph ? a word ?

I get that you extend marked with this https://marked.js.org/using_pro#use, and use this to close elements for text and list items. But I feel that we might miss something if there is the possibility of many unclosed elements.

Reply:
Very good question, let me shed some light on this.

The transformation is a two-stage process. The first pass is done by the "lexer" which takes the markdown and creates a tree of tokens. Any closed inline element is represented using a specific token. Any content that is not a known Markdown structure ends up being "text".

The second pass is done by the "parser" which takes the token tree and renders it to HTML. In this case, I'm relying on a [custom renderer](https://marked.js.org/using_pro#renderer) to try and find unclosed inline elements (that the lexer couldn't detect) and complete them.

To be fair, I tried adding the custom logic to the lexer, but didn't have much success manipulating the token tree.

To circle back to your original question, `text` can represent the content of a paragraph, a list item, a table cell, etc.

I didn't not experience issues with nested inline elements, but you are right that there could be some corner cases to handle there.

Reply:
Like we discussed outside this thread, maybe the model will never return stuff like bold in the middle of an italic text. Trying to fix those case would be like constructing a bridge over a river that doesn't exist. 

Thanks for the clarification, in all case !